### PR TITLE
Update install instructions

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

- Add conda install instructions
- PyGMT now works well with GMT on Windows
- Fixes a few documentation issues.
- List netcdf4 as a dependency

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

### CODE REVIEW

**Comment by seisman** [comment]
Put it to WIP. Will improve it after #434 is merged.

File: doc/install.rst [hunk 1]
```diff
@@ -134,7 +129,7 @@ This can happen if you have multiple versions of GMT installed.
 You can tell PyGMT exactly where to look for ``libgmt`` by setting the
 ``GMT_LIBRARY_PATH`` environment variable.
 This should be set to the directory where ``libgmt.so``, ``libgmt.dylib`` or ``gmt.dll``
-can be found for Linux, MacOS and Windows respectively.
+can be found for Linux, macOS and Windows respectively.
 e.g. in a terminal run::
 
    export GMT_LIBRARY_PATH=$HOME/anaconda3/envs/pygmt/lib
```
Comment (weiji14) [inline doc/install.rst:135]: Would be good to have the instructions for Windows here as well.

```suggestion
    # Linux/macOS
    export GMT_LIBRARY_PATH=$HOME/anaconda3/envs/pygmt/lib
    # Windows
    set "GMT_LIBRARY_PATH=C:\Miniconda\envs\pygmt\Library\bin\"
   
```

File: doc/install.rst [hunk 2]
```diff
@@ -134,7 +129,11 @@ This can happen if you have multiple versions of GMT installed.
...
 This should be set to the directory where ``libgmt.so``, ``libgmt.dylib`` or ``gmt.dll``
-can be found for Linux, MacOS and Windows respectively.
+can be found for Linux, macOS and Windows respectively.
 e.g. in a terminal run::
 
-   export GMT_LIBRARY_PATH=$HOME/anaconda3/envs/pygmt/lib
+    # Linux/macOS
+    export GMT_LIBRARY_PATH=$HOME/anaconda3/envs/pygmt/lib
+    # Windows
+    set "GMT_LIBRARY_PATH=C:\Miniconda\envs\pygmt\Library\bin\"
```
Comment (seisman) [inline doc/install.rst:138]: What about changing it to:
```suggestion
    set "GMT_LIBRARY_PATH=C:\Users\USERNAME\Anaconda3\envs\pygmt\Library\bin\"
```

Comment (seisman) [reply doc/install.rst:138]: On Windows, Anaconda is installed to `C:\Users\<your-username>\Anaconda3\` by default.

Comment (weiji14) [reply doc/install.rst:138]: Either way, as long as there's an example. The Anaconda one does look better though.

File: doc/install.rst [hunk 3]
```diff
@@ -79,7 +75,7 @@ First, we must configure conda to get packages from the
 Now we can create a new conda environment with Python and all our dependencies installed
 (we'll call it ``pygmt`` but you can change it to whatever you want)::
 
-     conda create --name pygmt python=3.6 pip numpy pandas xarray packaging gmt=6.0.0
+     conda create --name pygmt python=3.8 pip numpy pandas xarray packaging gmt=6.0.0
```
Comment (seisman) [inline doc/install.rst:78]:
```suggestion
     conda create --name pygmt python=3.8 pip numpy pandas xarray packaging gmt
```

Comment (weiji14) [reply doc/install.rst:78]: I'm fine with this. You'll get a merge conflict with the netcdf branch though. Do you want to just merge the netcdf one into this PR?

Comment (seisman) [reply doc/install.rst]: Yes, merged that branch in to make life easier.

**Pull request merged**